semeio/reporter/serialize_json.py

```diff
@@ -0,0 +1,10 @@
+import json
+
+
+def serialize_json(key, val):
+    try:
+        with open("{}.json".format(key), "w") as f:
+            json.dump(val, f)
+    except:
```

Comment:
This is bad for a number of reasons, but mainly to separate os errors (disk full, lacking permissions) and application errors (non-serializable things like dates etc).

Reply:
Note that the JSON module uses very generic errors. `TypeError` is thrown when the type is not JSON-serialisable, `OverflowError` is thrown when there's a cycle in the input.

One possibility for finer exception handling is to overload with our own `JSONEncoder.default` function that throws something other than `TypeError`, however, the docs state that:

> If specified, default should be a function that gets called for objects that can’t otherwise be serialized. It should return a JSON encodable version of the object or raise a TypeError. If not specified, TypeError is raised.

It seems that the doc is pretty strict about it this being a `TypeError`, and I'm not sure if we should break this contract.